docs/content/dagster-cloud/deployment/serverless.mdx

```diff
@@ -92,6 +91,49 @@ Often you'll need to securely access secrets from your jobs. With Dagster Cloud
 - If you're using our GitHub integration, you can pass in secrets as a JSON string input to the serverless deploy step in your repository's Github action script.
 - If you're using the CLI directly, you can pass `KEY=VALUE` pairs using the `--env` flag. Run `dagster-cloud serverless deploy --help` to learn more.
 
+### Adding dependencies
+
+Any dependencies specified in either `requirements.txt` or `setup.py` will be installed for you automatically by the Dagster Cloud Serverless infrastructure.
+
+---
+
+## Customizing the container
+
+Many apps will work fine with the default Dagster Cloud Serverless setup. However, some apps may need to make changes to the base image, either to use a different OS, different Python version, or install some native dependencies. You can customize your image through our Serverless lifecycle hooks, and/or changing the base image.
+
+### Lifecycle hooks
+
+This method is the easiest to set up, and does not require setting up any additional infrastructure.
+
+In the root of your repo, you can provide two optional shell scripts: `dagster_cloud_pre_install.sh` and `dagster_cloud_post_install.sh`. These will run before and after Python dependencies are installed. They are useful for installing any non-Python dependencies or otherwise configuring your environment.
+
+### Changing the base image
+
+This method is the most flexible, but requires setting up a pipeline outside of Dagster to build a custom base image.
+
+The default base image is `debian:bullseye-slim`, but it can be changed.
+
+- **With GitHub**: you can provide a `base_image` input parameter to the "build and deploy" step in your GitHub Actions configuration (usually at `.github/workflows/deploy.yml`):
+
+```
+- name: Build and deploy to Dagster Cloud serverless
+  uses: dagster-io/dagster-cloud-action/actions/serverless_prod_deploy@v0.1
+  with:
+    dagster_cloud_api_token: ${{ secrets.DAGSTER_CLOUD_API_TOKEN }}
+    location: ${{ toJson(matrix.location) }}
+    # Use a custom base image
+    base_image: "my_base_image:latest"
+    organization_id: ${{ secrets.ORGANIZATION_ID }}
+  env:
+    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+```
+
+- **With the CLI**: add the `--base-image` CLI argument to the deploy command to specify the registry path to the desired base image.
+
+```
```

Comment:
Same comments here as for the code block under the GitHub/previous bullet point. To get syntax highlighting + consistent formatting, this would be:
```
```shell (or bash, whichever you feel is more correct)
```